pkg/apis/extensions/validation/network.go

```diff
@@ -100,3 +106,42 @@ func ValidateNetworkStatusUpdate(newStatus, oldStatus *extensionsv1alpha1.Networ
 
 	return allErrs
 }
+
+var availableIPFamilies = sets.NewString(
+	string(extensionsv1alpha1.IPFamilyIPv4),
+	string(extensionsv1alpha1.IPFamilyIPv6),
+)
+
+// ValidateIPFamilies validates the given list of IP families for valid values and combinations.
+func ValidateIPFamilies(ipFamilies []extensionsv1alpha1.IPFamily, fldPath *field.Path) field.ErrorList {
```

Comment:
Yes, we also tried deduplicating this function. 
Our best guess was using a generic function that accepts slices of both `core.IPFamily` and `extensionsv1alpha1.IPFamily`.
This would need to be placed in a new common helper package to prevent cyclic dependencies.
Also, this would need to use another set of constants (different constants than in the API packages).
All of this was rather messy, unreadable, and based on multiple assumptions.
We didn't like this approach and went back to what is currently done in this PR.

Considering, that the validation logic does derivate and might further drift in the future and that other validation/defaulting logic is also highly duplicated between API groups/versions, we felt this is the way to go.

If you object, please provide an example of how you would like to achieve the duplication.

Reply:
I see, then let's keep it as it is.